Modules/_functoolsmodule.c

```diff
@@ -399,13 +401,32 @@ partial_repr(partialobject *pto)
         if (arglist == NULL)
             goto done;
     }
-    result = PyUnicode_FromFormat("%s(%R%U)", Py_TYPE(pto)->tp_name,
-                                  pto->fn, arglist);
+
+    mod = _PyType_GetModuleName(Py_TYPE(pto));
+    if (mod == NULL) {
+        goto error;
+    }
+    else {
+        name = PyType_GetQualName(Py_TYPE(pto));
+        if (name == NULL) {
+            Py_DECREF(mod);
+            goto error;
+        }
+        result = PyUnicode_FromFormat("%U.%S(%R%U)", mod, name, pto->fn,
```

Comment:
```suggestion
        result = PyUnicode_FromFormat("%S.%S(%R%U)", mod, name, pto->fn,
```
It is not guaranteed that `__module__` is a string.